File: SGPla/Models/EntidadAcademica.cs
using System;
using System.Collections.Generic;

namespace SGPla.Models;

public partial class EntidadAcademica
{
    public int IdEntidadAcademica { get; set; }

    public int IdAreaAcademica { get; set; }

    public string Nombre { get; set; } = null!;

    public string CalleNumero { get; set; } = null!;

    public string Colonia { get; set; } = null!;

    public string Cp { get; set; } = null!;

    public string Municipio { get; set; } = null!;

    public string Telefono { get; set; } = null!;

    public string Conmutador { get; set; } = null!;

    public string Extension { get; set; } = null!;

    public string Fax { get; set; } = null!;

    public string Region { get; set; } = null!;

    public string Campus { get; set; } = null!;

    public virtual ICollection<Aviso> Aviso { get; set; } = new List<Aviso>();

    public virtual ICollection<CoordinadorEa> CoordinadorEa { get; set; } = new List<CoordinadorEa>();

    public virtual AreaAcademica IdAreaAcademicaNavigation { get; set; } = null!;

    public virtual ICollection<IntegranteCt> IntegranteCt { get; set; } = new List<IntegranteCt>();

    public virtual ICollection<ProgramaEducativo> ProgramaEducativo { get; set; } = new List<ProgramaEducativo>();
}

File: sgpla-app/src/modules/entidades-academicas/domain/entidad-academica.ts
export const REGIONES = [
  "1-Xalapa",
  "2-Veracruz",
  "3-Orizaba-Córdoba",
  "4-Poza Rica-Túxpan",
  "5-Coatzacoalcos-Minatitlán",
] as const;

export type Region = (typeof REGIONES)[number];

export type EntidadAcademica = {
  idEntidadAcademica: number;
  clave: string | null;
  nombre: string;
  calleNumero: string;
  colonia: string;
  cp: string;
  municipio: string;
  telefono: string;
  extension: string;
  idAreaAcademica: number;
  nombreAreaAcademica: string;
  region: string;
};

export type Pagina<T> = {
  items: T[];
  pagina: number;
  cantidad: number;
  total: number;
};

export type ConsultarEntidadesAcademicasQuery = {
  busqueda?: string;
  region?: string;
  idAreaAcademica?: number;
  pagina: number;
  cantidad: number;
};

export type GuardarEntidadAcademicaInput = {
  clave: string;
  nombre: string;
  calleNumero: string;
  colonia: string;
  cp: string;
  municipio: string;
  telefono: string;
  extension: string;
  idAreaAcademica: number;
  region: string;
};

export type AreaAcademica = {
  idAreaAcademica: number;
  nombre: string;
  telefono: string;
  extension: string;
};
